apps/dashboard/src/components/workflow-list.tsx

```
@@ -158,7 +160,18 @@ export const WorkflowList = () => {
                             <FaCode className="size-3" />
                           </Badge>
                         )}
-                        <TruncatedText text={workflow.name} />
+                        <TruncatedText
+                          className="cursor-pointer"
+                          text={workflow.name}
+                          onClick={() => {
+                            navigate(
+                              buildRoute(ROUTES.EDIT_WORKFLOW, {
+                                environmentId: currentEnvironment?._id ?? '',
```

This is pure inner DX but also a way to prevent invalid states in the front end, such as a workflow route, without environmentId. 

Can we please update the hook times so that they always return a currentEnvironment when used to avoid this ternary on each current environment?

There are two ways to go about this.

1. The hook with throw an error if used and the environment is not set (fail fast), but the environment return type will always be `IEnvironment`.

2. We can teach the hook to return the `{isLoaded, currentEnvironment}` tuple and use discriminated unions so that if `isLoaded` is true currentEnvironment is always an IEnvironment. Then, in all the components that use `useEnvironment` we can do `if (!isLoaded) { return null }`

---

@SokratisVidros, it's not that simple:
1. Throwing an error won't work, because on the initial render the `currentEnvironment` is always undefined, so the `useEnvironment` hook will always throw.
2. That's a partial solution to this problem because all code above the `if (!isLoaded)` including hooks will still need to check whether the current env is present.

---

We will take this offline. It's not a good enough blocker for this PR.